lib/shared-method.js

```diff
@@ -231,6 +227,26 @@ SharedMethod.prototype.invoke = function (scope, args, fn) {
         }
       }
 
+      //Is the arg a valid value?
+      if ((desc.required || actualType !== 'undefined')) {
+        var errFlag = 0;
+        var err = new Error('\'' + uarg + '\' is not one of: ');
+
+        if (desc.enum && !(desc.enum.indexOf(uarg) > -1)) {
+          err.message += desc.enum.join(", ");
+          errFlag = 1;
+        }
+        else if (desc.items && desc.items.enum && !(desc.items.enum.indexOf(uarg) > -1)) {
```

Comment:
Won't `uarg` be an `array`?  I'd expect you to have to iterate over the contents of `uarg`, checking if each entry is in `enum`